plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -81,14 +94,24 @@ export async function createRouter(
       });
     })
     .post('/v1/jobs', async (req, res) => {
-      const template: TemplateEntityV1alpha1 = req.body.template;
+      const templateName: string = req.body.templateName;
       const values: RequiredTemplateValues & Record<string, JsonValue> =
         req.body.values;
 
+      let template: TemplateEntityV1alpha1;
+      try {
+        template = await entityClient.findTemplate(templateName);
+      } catch (e) {
+        // help me here
+        res.status(400).json({ errors: e });
```

Comment:
Review this one.

Reply:
I think if you remove the try catch, there's middleware in express that will translate the errors thrown from the API into the error messages.

Reply:
Could it be argued that the client shouldn't throw, but rather return undefined for not-found and let the API layer deal with how to communicate that?

Either way yes, throwing an InputError or NotFoundError does get translated accordingly https://github.com/spotify/backstage/blob/master/packages/backend-common/src/errors.ts#L53

Final note: not sure you should be returning json with an Error instance in it right? If you want to serialize it as a string make sure to grab .message or .toString() it, for example

Reply:
+1 for letting the middleware take care of the serialization

Reply:
Thanks  😺